code/game/objects/items/devices/pressureplates.dm

```diff
@@ -59,21 +60,28 @@
 
 /obj/item/pressure_plate/proc/trigger()
 	can_trigger = TRUE
-	if(istype(sigdev))
-		sigdev.signal()
+	if(istype(assembly))
+		assembly.activate()
 
-/obj/item/pressure_plate/attackby(obj/item/I, mob/living/L)
-	if(issignaler(I) && !istype(sigdev) && removable_signaller && L.transferItemToLoc(I, src))
-		sigdev = I
-		to_chat(L, span_notice("You attach [I] to [src]!"))
+/obj/item/pressure_plate/attackby(obj/item/item, mob/living/L)
+	if(isassembly(item) && !istype(assembly) && removable_assembly)
+		var/obj/item/assembly/new_assembly = item
+		if(!(new_assembly.assembly_behavior & ASSEMBLY_FUNCTIONAL_OUTPUT))
+			to_chat(L, span_warning("\The [item] doesn't seem like it would do much of anything inside of [src]..."))
+			return
+		if(L.transferItemToLoc(item, src))
+			assembly = item
+			SEND_SIGNAL(item, COMSIG_ASSEMBLY_ADDED_TO_OBJECT, src, L)
```

Comment:
See above, this would become COMSIG_ASSEMBLY_ADDED_TO_PRESSURE_PLATE.

Alternatively, this behaviour would be abstracted to something else like a component or base type and handled there so that you only need to call this signal once, but since you are calling it from multiple points in code, it needs to be distinct from one another.